### PR TITLE
Initial Java Upgrade changes

### PR DESCRIPTION
**Pull request by foman-ch**
**JIRA**: https://companieshouse.atlassian.net/browse/JU-385

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -137,11 +159,16 @@
             <groupId>org.springframework.kafka</groupId>
             <artifactId>spring-kafka-test</artifactId>
             <scope>test</scope>
+            <version>3.0.13</version>
```
Comment (dgurjar-ch) [inline pom.xml:162]: Please define version in properties section.

File: src/main/java/uk/gov/companieshouse/itemgroupworkflowapi/model/ItemDescriptionIdentifier.java [hunk 2]
```diff
@@ -17,7 +17,7 @@ public enum ItemDescriptionIdentifier {
             .collect(Collectors.toMap(ItemDescriptionIdentifier::toString, Function.identity()));
     }
 
-    private ItemDescriptionIdentifier(String identifierName) {
+    ItemDescriptionIdentifier(String identifierName) {
```
Comment (dgurjar-ch) [inline src/main/java/uk/gov/companieshouse/itemgroupworkflowapi/model/ItemDescriptionIdentifier.java:20]: Please define access identifier, make it protected

File: src/main/java/uk/gov/companieshouse/itemgroupworkflowapi/model/ItemKind.java [hunk 3]
```diff
@@ -17,7 +17,7 @@ public enum ItemKind {
             .collect(Collectors.toMap(ItemKind::toString, Function.identity()));
     }
 
-    private ItemKind(String itemKindName) {
+    ItemKind(String itemKindName) {
```
Comment (dgurjar-ch) [inline src/main/java/uk/gov/companieshouse/itemgroupworkflowapi/model/ItemKind.java:20]: Same , please define access modifier

File: pom.xml (hunk 1 above)
Comment (foman-ch) [reply pom.xml]: good spot, thanks

File: src/main/java/uk/gov/companieshouse/itemgroupworkflowapi/model/ItemKind.java (hunk 3 above)
Comment (foman-ch) [reply src/main/java/uk/gov/companieshouse/itemgroupworkflowapi/model/ItemKind.java:20]: same as above, adding protected gives a compile error

File: src/main/java/uk/gov/companieshouse/itemgroupworkflowapi/model/ItemDescriptionIdentifier.java (hunk 2 above)
Comment (hwatkins-ch) [reply src/main/java/uk/gov/companieshouse/itemgroupworkflowapi/model/ItemDescriptionIdentifier.java:20]: Every enum constructor without an access modifier is private. 
Using the public or protected modifiers will cause a compilation error. It's only possible to define private constructors on an enum type without causing compilation errors.

**Review by hwatkins-ch — APPROVED** [review]

**Review by dgurjar-ch — APPROVED** [review]

**Pull request merged**